### DISCUSSION TITLE
Retrieve files from LittleFS

### DISCUSSION THREAD

**Discussion by econeale** [espressif/arduino-esp32#7776] [Q&A]
Does anyone know if there's an "easy" way to retrieve files from the SPIFFS / LittleFS without using a Webserver or similar? 

For V1 of the Arduino IDE, @me-no-dev had [this](https://github.com/me-no-dev/arduino-esp32fs-plugin) handy tool for uploading data directories to the device but I don't think there's been a way to get that file from the ESP32 to the host via esptool or similar. 

All thoughts appreciated!

**Comment by mrengineer7777** [comment]
I use the filesystem upload tool in VSCode + PlatformIO.  I believe it uses esptool to write the partition to the ESP32, typically over USB bootloader.  Anything that involves reading an individual file will need to have code running on the ESP32 and some communications channel.
